ddprof-exporter/src/tag.rs

```diff
@@ -65,9 +65,67 @@ impl Tag {
     }
 }
 
+pub fn parse_tag_chunk<'a>(chunk: &'a str) -> Result<Tag, Cow<'static, str>> {
+    if let Some(first_colon_position) = chunk.find(':') {
+        // A tag which leads with a colon isn't explicitly invalid, but what
+        // are we supposed to do about it?
+        if first_colon_position == 0 {
+            return Err("tag cannot start with a colon".into());
+        }
+        if let Some(last_char) = chunk.chars().last() {
+            if last_char == ':' {
```

Comment:
Minor: I think this can be folded into 
```suggestion
        if let Some(':') = chunk.chars().last() {
```